diskmaker/controllers/lv/reconcile.go

```diff
@@ -143,13 +168,38 @@ func (r *LocalVolumeReconciler) createSymlink(
 	successMsg := fmt.Sprintf("found matching disk %s with id %s", deviceNameLocation.diskNamePath, deviceNameLocation.diskID)
 	r.eventSync.Report(r.localVolume, newDiskEvent(FoundMatchingDisk, successMsg, deviceNameLocation.diskNamePath, corev1.EventTypeNormal))
 	return true
-
 }
+
 func diskMakerLabels(crName string) map[string]string {
 	return map[string]string{
 		"app": fmt.Sprintf("local-volume-diskmaker-%s", crName),
 	}
 }
+
+func (r *LocalVolumeReconciler) checkForExistingPV(deviceNameLocation DiskLocation) bool {
+	pvs := r.runtimeConfig.Cache.ListPVs()
+	nodeLabels := r.runtimeConfig.Node.GetLabels()
+
+	hostname, found := nodeLabels[corev1.LabelHostname]
+	if !found {
+		return false
+	}
+
+	deviceName := filepath.Base(deviceNameLocation.diskNamePath)
+
+	for i := range pvs {
+		pv := pvs[i]
```

Comment:
Minor nit: you could use `for _, pv := range pvs` like you did on line 276 and avoid the extra `pv := pvs[i]` line. Purely optional.

Reply:
I was doing that on purpose to avoid globbering of loop variables by goroutines. I know that is not the case here, but I am trying to adopt this pattern as matter of style rather than only when needed.

Reply:
Ok that's fine, just caught my eye.